src/uct/ib/base/ib_device.c

```diff
@@ -1125,9 +1125,10 @@ ucs_status_t uct_ib_device_mtu(const char *dev_name, uct_md_h md, int *p_mtu)
     return UCS_OK;
 }
 
-int uct_ib_device_is_gid_raw_empty(uint8_t *gid_raw)
+int uct_ib_device_is_invalid_gid(uint8_t *gid_raw)
 {
-    return (*(uint64_t *)gid_raw == 0) && (*(uint64_t *)(gid_raw + 8) == 0);
+    /* check if the lower 64 bits are zero, which means the GID is invalid */
+    return (*(uint64_t *)(gid_raw + 8) == 0);
```

Comment:
maybe type cast to "union ibv_gid" and check the guid field? instead of pointer arithmetic

Reply:
done